FlowCrypt/Functionality/Services/GoogleUserService.swift

```diff
@@ -164,46 +157,26 @@ extension GoogleUserService {
         return GTMAppAuthFetcherAuthorization(fromKeychainForName: Constants.index + email)
     }
 
+    // todo - isn't this call supported by Google client library?
     private func fetchGoogleUser(
-        with authorization: GTMAppAuthFetcherAuthorization?,
-        completion: @escaping ((Result<GoogleUser, GoogleUserServiceError>) -> Void)
-    ) {
-        guard let authorization = authorization else {
-            fatalError("authorization should not be nil at this point")
-        }
-
-        guard let userInfoEndpoint = URL(string: "https://www.googleapis.com/oauth2/v3/userinfo") else {
-            fatalError("userInfoEndpoint could not be nil because it's hardcoded string url")
+        with authorization: GTMAppAuthFetcherAuthorization
+    ) async throws -> GoogleUser {
+        guard let url = URL(string: Constants.userInfoUrl) else {
+            throw AppErr.unexpected("URL(Constants.userInfoUrl) nil")
         }
-
         let fetcherService = GTMSessionFetcherService()
         fetcherService.authorizer = authorization
-
-        fetcherService.fetcher(with: userInfoEndpoint)
-            .beginFetch { data, error in
-                if let data = data {
-                    do {
-                        let user = try JSONDecoder().decode(GoogleUser.self, from: data)
-                        completion(.success(user))
-                    } catch {
-                        completion(.failure(.parsingError(error)))
-                    }
-                } else if let error = error {
-                    completion(.failure(.serviceError(error)))
-                } else {
-                    completion(.failure(.inconsistentState("Fetching user")))
-                }
-            }
-    }
-
-    private func handleUserInfo(error: Error) {
-        if (error as NSError).isEqual(OIDOAuthTokenErrorDomain) {
-            logger.logError("Authorization error during token refresh, clearing state. \(error)")
-            if let email = currentUserEmail {
+        do {
+            let data = try await fetcherService.fetcher(with: url).beginFetch()
+            return try JSONDecoder().decode(GoogleUser.self, from: data)
+        } catch {
+            let isTokenErr = (error as NSError).isEqual(OIDOAuthTokenErrorDomain)
+            if isTokenErr, let email = self.currentUserEmail {
+                self.logger.logError("Authorization error during token refresh, clearing state. \(error)")
+                // todo - what exactly does this do and why?
```

Comment:
It removes any authorisation information which was stored in Keychain, the same happens on logout.
So if any error happens during token refresh then user will be signed out automatically.